Makefile

```diff
@@ -22,6 +22,89 @@ KIND_CLUSTER_NAME ?= capi-test
 # ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
 ENVTEST_K8S_VERSION = 1.23
 
+#
+# Directories.
+#
+# Full directory of where the Makefile resides
+ROOT_DIR:=$(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
+REPO_ROOT := $(shell git rev-parse --show-toplevel)
+EXP_DIR := exp
+BIN_DIR := bin
+TEST_DIR := test
+TOOLS_DIR := $(REPO_ROOT)/hack/tools
+TOOLS_BIN_DIR := $(abspath $(TOOLS_DIR)/$(BIN_DIR))
+E2E_FRAMEWORK_DIR := $(TEST_DIR)/framework
+CAPD_DIR := $(TEST_DIR)/infrastructure/docker
+GO_INSTALL := $(REPO_ROOT)/scripts/go_install.sh
```

Comment:
why we stop using the installation logic from the Makefile himself and put it in a side script ?

Reply:
we were reinventing wheel by doing our own way when we can just keep it same as cluster-api repo way so that its easy maintain

Reply:
it seems it come from the operator-sdk, It was part to the original repo
no problem to change but let's clean the Makefile and remove all the go-get-tool items